### PR TITLE
Add ou:industryname and migrate props. Add it:prod:softname and migrate props. Add ou:team and refs. Add crypto:key / crypto:algorithm. Add crypto:currency:address:seed and crypto:smart:effect:* to model smart contract effects.

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/models/crypto.py [hunk 1]
```diff
@@ -34,6 +34,20 @@ def getModelDefs(self):
                 ('crypto:smart:contract', ('guid', {}), {
                     'doc': 'A smart contract.',
                 }),
+                ('crypto:smart:effect:transfertoken', ('guid', {}), {
```
Comment (reign-git) [inline synapse/models/crypto.py:37]: The name is close to "transfertokens" and may get confusing. Recommend considering another secondary prop name like "transferowner".

Comment (invisig0th) [reply synapse/models/crypto.py:37]: It's more about what you are transferring the ownership *of*.  This specific effect transfers 1 non-fungible token and the other transfers N fungible tokens. Open to name changes but "transferowner" is too general.

File: synapse/models/crypto.py [hunk 2]
```diff
@@ -34,6 +34,20 @@ def getModelDefs(self):
...
+                    'doc': 'A smart contract effect which transfers ownership of a non-fungible token.',
+                }),
+                ('crypto:smart:effect:transfertokens', ('guid', {}), {
+                    'doc': 'A smart contract effect which transfers fungible tokens.',
+                }),
+                ('crypto:smart:effect:supplytokens', ('guid', {}), {
```
Comment (reign-git) [inline synapse/models/crypto.py:43]: Supply makes me think tokens are being added or increased only. Thoughts on creating two forms to represent this concept - crypto:smart:effect:supplyincr and crypto:smart:effect:supplydecr for supply increase and decrease?

File: synapse/models/crypto.py [hunk 3]
```diff
@@ -92,7 +102,13 @@ def getModelDefs(self):
                 ('rsa:key', ('comp', {'fields': (('mod', 'hex'), ('pub:exp', 'int')), }), {
                     'doc': 'An RSA keypair modulus and public exponent.'
                 }),
-
+                ('crypto:key', ('guid', {}), {
+                    'doc': 'A cryptographic key and algorithm.',
+                }),
+                ('crypto:algorithm', ('str', {'lower': True, 'onespace': True, 'strip': True}), {
```
Comment (Cisphyx) [inline synapse/models/crypto.py:108]: strip is redundant when onespace is specified

File: synapse/models/infotech.py [hunk 4]
```diff
@@ -387,7 +375,10 @@ def getModelDefs(self):
                     'doc': 'A Windows registry key/value pair.',
                 }),
                 ('it:prod:soft', ('guid', {}), {
-                    'doc': 'A arbitrary, unversioned software product.',
+                    'doc': 'A software product.',
+                }),
+                ('it:prod:softname', ('str', {'strip': True, 'onespace': True, 'lower': True}), {
```
Comment (Cisphyx) [inline synapse/models/infotech.py:380]: redundant strip

File: synapse/models/orgs.py [hunk 5]
```diff
@@ -47,6 +50,9 @@ def getModelDefs(self):
                 ('ou:industry', ('guid', {}), {
                     'doc': 'An industry classification type.',
                 }),
+                ('ou:industryname', ('str', {'lower': True, 'strip': True, 'onespace': True}), {
```
Comment (Cisphyx) [inline synapse/models/orgs.py:53]: redundant strip

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**